packages/eslint-plugin/tests/rules/no-unsafe-type-assertion.test.ts

```diff
@@ -0,0 +1,559 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-unsafe-type-assertion';
+import { getFixturesRootDir } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  languageOptions: {
+    parserOptions: {
+      project: './tsconfig.noImplicitThis.json',
+      projectService: false,
+      tsconfigRootDir: getFixturesRootDir(),
+    },
+  },
+});
+
+ruleTester.run('no-unsafe-member-access', rule, {
+  valid: [
+    `
+declare const a: string;
+const b = a as string | number;
+    `,
+    `
+declare const a: string;
+const b = a as unknown;
+    `,
+    `
+declare const a: () => boolean;
+const b = a() as boolean | number | string;
+    `,
+    `
+declare const a: string;
+const b = a as string;
+    `,
+    `
+declare const a: string;
+const b = <string | number>a;
```

Comment:
Each test is here on two forms: `a as string` and `<string>a`. I did this manually rather than provide a function to do this. Do you think it's necessary to test each case both ways? Should I do this differently?